File: src/main/java/easy/CanMakeArithmeticProgression.java
package easy;

import java.util.Arrays;

public class CanMakeArithmeticProgression {
  public boolean solve(int[] arr) {
    Arrays.sort(arr);
    int index = 0;
    int distance = arr[0] - arr[1];
    while (index < arr.length - 1) {
      if (arr[index] - arr[index + 1] != distance) return false;
      index++;
    }
    return true;
  }
}

File: src/easy/can-make-arithmetic-progression.ts
export const canMakeArithmeticProgression = (arr: number[]): boolean => {
	arr.sort((a, b) => a - b);
	const distance = arr[0] - arr[1];
	for (let i = 1; i < arr.length - 1; i++) {
		if (arr[i] - arr[i + 1] !== distance) return false;
	}
	return true;
};
